File: Linear Search/LinearSearch.js
/*
La seguente funzione trova la prima occorrenza del valore v passato come argomento
Parametro -> a = L'array in cui si vuole effettuare la ricerca
Parametro -> v = valore di cui si vuole sapere l'indice nell'array
Return ->  l'indice della prima occorrenza di v oppure null se v non è presente nell'array a
*/
function linearSearch(a, v){
    for(let i = 0; i<a.length; i++)
        if(a[i] == v)
            return i;   
    return null;
}

module.exports = {
    linearSearch
}

File: Linear Search/LinearSearch.py
def linearSearch(a, v):
    for i in range(0, len(a)):
        if a[i] == v:
            print('a[%i]=%i == %i' % (i, a[i], v)) 
            return i
        else:
           print('a[%i]=%i != %i' % (i, a[i], v)) 

array = [6, 5, 4, 3, 2, 1]
linearSearch(array, 3)